delegator/src/sign/sign.service.ts

```diff
@@ -118,22 +102,18 @@ export class SignService {
   }
 
   async validateTransaction(tx) {
-    const result = await this.prisma.contract.findMany({
-      where: {
-        address: tx.to,
-        Reporter: {
-          some: {
-            address: tx.from
-          }
-        },
-        Function: {
-          some: {
-            encodedName: tx.input.substring(0, 10)
-          }
-        }
+    const encodedName = tx.input.substring(0, 10)
+    const contract = await this.prisma.contract.findUnique({
+      where: { address: tx.to },
+      include: {
+        reporter: { where: { address: tx.from } },
+        function: { where: { encodedName } }
       }
     })
-    if (result.length == 0) {
+
+    if (contract && contract.reporter.length != 0 && contract.function.length != 0) {
```

Comment:
I think we do not need to test on `contract.reporter.length != 0 && contract.function.length != 0`

Reply:
The previous approach of how I filter query was not correct. 

I fixed the issue, and here is the correct way of how to filter query from relationships. 

```Typescript
      const relationQuery = await this.prisma.contract.findMany({
      where: {
        address: tx.to,
        reporter: { some: { address: tx.from } },
        function: { some: { encodedName } }
      },
      include: {
        reporter: { where: { address: tx.from } },
        function: { where: { encodedName } }
      }
    })
```

Reply:
Great! ❤️